src/components/AddItem/AddItem.js

```diff
@@ -4,37 +4,82 @@ import NavBar from '../NavBar/NavBar';
 
 const AddItem = () => {
   const [groceryItem, setGroceryItem] = useState('');
-  const [purchaseDate, setPurchaseDate] = useState(null);
   const [howSoon, setHowSoon] = useState(null);
-  const listData = firestore.collection('groceryItems');
+  // const listData = firestore.collection('groceryItems'); commented out due to not wanting to direct to the 'groceryItems' collection but use the entire database instead
 
   const itemInputChange = (e) => {
     setGroceryItem(e.target.value);
   };
 
-  const dateInputChange = (e) => {
-    setPurchaseDate(e.target.value);
-  };
   const radioInputChange = (e) => {
     setHowSoon(parseInt(e.target.value));
     console.log(howSoon);
   };
 
+  const cleanUserInput = (item) => {
+    // Step 1: Sanitizing and trying out the regular expression symbols and converting to lowercase
+    return item.replace(/[\W_]+/g, '').toLowerCase();
+  };
+
   const handleSubmitClick = async (e) => {
     e.preventDefault();
-    await listData.add({
-      // writing a new document to firestore with the select values/fields
+
+    const existingItem = await duplicateItem(groceryItem); // Step 2: I'm trying to add a new function as a addon that will check for dupe items with an alert
+    if (existingItem) {
+      // **== I'm stuck on trying to get it to work!! ===**
+      alert('The Item already exists!');
+
+      setGroceryItem('');
+      setHowSoon(null);
+      return;
+    }
+
+    firestore.collection(localStorage.getItem('token')).add({
+      // new format that allows storage of a item under the right token
+      // writing a new document to firestore with these values/fields
       name: groceryItem,
-      frequency: howSoon,
-      token: window.localStorage.getItem('token'),
+      howSoon: howSoon,
       createdAt: Date.now(),
-      purchaseDate: purchaseDate,
+      purchaseDate: null,
     });
 
     setGroceryItem('');
     setHowSoon(null);
   };
 
+  const duplicateItem = async (item) => {
+    let existingItem = false;
+    console.log('token:', localStorage.getItem('token'));
+    await firestore // this is new syntax that can apply the sanitize to each individual item using 'querySnapshot'
+      .collection(localStorage.getItem('token'))
+      .get()
+      .then((querySnapshot) => {
+        querySnapshot.forEach((doc) => {
+          /*console.log(
+            'doc.data().name: ',
+            doc.data().name,
+
+            cleanUserInput(doc.data().name) === cleanUserInput(item),
+        );*/
+
+          // the 'doc' was a property from the firestore docs to I'm trying it out here
+          if (
+            // created this if statement to target and apply sanitizer to the user input
+            cleanUserInput(doc.data().name) === cleanUserInput(item)
```

Comment:
I see two things here that could be improved:

1. It's more recommended to leave the comment out of the `if` statement and keep the condition in the same line as the statement (`if`).
```suggestion
        // created this if statement to target and apply sanitizer to the user input
        if (cleanUserInput(doc.data().name) === cleanUserInput(item)) {
            existingItem = true;
        }
```

2. `cleanUserInput(item)` can be stored in a variable outside of the `forEach`. The way it currently works, you are sanitizing the same `item` value with every document in the collection, so it's better if you do it only once, then reference that value inside here. [**NOTE:** No need to change if you don't see this add any value to the current implementation 😀]

```js
// at the beginning of `duplicateItem`
const sanitizedItem = cleanUserInput(item);
...
if (cleanUserInput(doc.data().name) === sanitizedItem) {
...
```